server/src/main/java/org/elasticsearch/bootstrap/PolicyUtil.java

```diff
@@ -23,31 +23,170 @@
+            // write is needed because of HdfsPlugin
+            new PropertyPermission("*", "read,write"),
+            new AuthPermission("doAs"),
+            new AuthPermission("doAsPrivileged"),
+            new AuthPermission("getSubject"),
+            new AuthPermission("getSubjectFromDomainCombiner"),
+            new AuthPermission("setReadOnly"),
+            new AuthPermission("modifyPrincipals"),
+            new AuthPermission("modifyPublicCredentials"),
+            new AuthPermission("modifyPrivateCredentials"),
+            new AuthPermission("refreshCredential"),
+            new AuthPermission("destroyCredential"),
+            new AuthPermission("createLoginContext.*"),
+            new AuthPermission("getLoginConfiguration"),
+            new AuthPermission("setLoginConfiguration"),
+            new AuthPermission("createLoginConfiguration.*"),
+            new AuthPermission("refreshLoginConfiguration")
+        );
+        // While it would be ideal to represent all allowed permissions with concrete instances so that we can
+        // use the builtin implies method to match them against the parsed policy, this does not work in all
+        // cases for two reasons:
+        // (1) Some permissions classes do not have a name argument that can represent all possible variants.
+        //     For example, FilePermission has "<< ALL FILES >>" so all paths can be matched, but DelegationPermission
+        //     does not have anything to represent all principals.
+        // (2) Some permissions classes are in java modules that are not accessible from the classloader used by
+        //     the policy parser. This results in those permissions being in UnresolvedPermission instances. Those
+        //     are normally resolved at runtime when that permission is checked by SecurityManager. But there is
+        //     no general purpose utility to resolve those permissions, so we must be able to match those
+        //     unresolved permissions in the policy by class and name values.
+        // Given the above, the below map is from permission class to the list of allowed name values. A sentinel value
+        // is used to mean names are accepted. We do not use this model for all permissions because many permission
+        // classes have their own meaning for some form of wildcard matching of the name, which we want to delegate
+        // to those permissions if possible.
+        Map<String, List<String>> classPermissions = Map.of(
+            URLPermission.class, ALLOW_ALL_NAMES,
+            DelegationPermission.class, ALLOW_ALL_NAMES,
+            ServicePermission.class, ALLOW_ALL_NAMES,
+            PrivateCredentialPermission.class, ALLOW_ALL_NAMES,
+            SQLPermission.class, List.of("callAbort", "setNetworkTimeout"),
+            ClassPermission.class, ALLOW_ALL_NAMES
+        ).entrySet().stream().collect(Collectors.toMap(e -> e.getKey().getCanonicalName(), Map.Entry::getValue));
```

Comment:
Is there a reason we don't just build the map with the canonical class name?

```suggestion
        Map<String, List<String>> classPermissions = Map.of(
            URLPermission.class.getCanonicalName(), ALLOW_ALL_NAMES,
            DelegationPermission.class.getCanonicalName(), ALLOW_ALL_NAMES,
            ServicePermission.class.getCanonicalName(), ALLOW_ALL_NAMES,
            PrivateCredentialPermission.class.getCanonicalName(), ALLOW_ALL_NAMES,
            SQLPermission.class.getCanonicalName(), List.of("callAbort", "setNetworkTimeout"),
            ClassPermission.class.getCanonicalName(), ALLOW_ALL_NAMES
        );
```

Reply:
Just more cumbersome to add. I found it easier to do the common manipulation at the end on all the entries.